planar_robot:
    # Publish all joint states -----------------------------------
    joint_state_controller:
        type: joint_state_controller/JointStateController
        publish_rate: 50  

    # Joint 1 position controller --------------------------------
    joint_1_position_controller:
        joint: joint_1
        pid:
            p: 100.0
            i: 15.0
            d: 20
        type: effort_controllers/JointPositionController

    # Joint 2 position controller --------------------------------
    joint_2_position_controller:
        joint: joint_2
        pid:
            p: 100.0
            i: 15.0
            d: 20
        type: effort_controllers/JointPositionController

    joint_trajectory_controller:
        type: effort_controllers/JointTrajectoryController
        joints:
            - joint_1
            - joint_2
        constraints:
            goal_time: 1.0
            stopped_velocity_tolerance: 0.5
        stop_trajectory_duration: 1.0
        state_publish_rate:  25
        action_monitor_rate: 25
        gains:
            joint_1: {p: 100.0, i: 15.0, d: 20.0, i_clamp_min: -5.0, i_clamp_max: 5.0}
            joint_2: {p: 100.0, i: 15.0, d: 20.0, i_clamp_min: -5.0, i_clamp_max: 5.0}
